plugins/org.jkiss.dbeaver.net.ssh.sshj/src/org/jkiss/dbeaver/model/net/ssh/SSHImplementationSshj.java

```diff
@@ -86,14 +87,17 @@ protected synchronized void setupTunnel(
 
             try {
                 if (index > 0) {
-                    final int port = setPortForwarding(clients[index - 1], host.getHostname(), host.getPort());
+                    final SSHClient prevClient = clients[index - 1];
```

Comment:
`setPortForwarding` method was used here, but it contains not only socket creation. We miss `LocalPortListener` by replacing this with `newDirectConnection` method. Don't we need it?
``` 
setPortForwarding(...) {
        ...
        final LocalPortForwarder forwarder = client.newLocalPortForwarder(parameters, ss);
        final LocalPortListener listener = new LocalPortListener(forwarder, parameters);
        listener.start();
        listeners.add(listener);
        ...
}
```

Reply:
It looks like it's not used anymore and, therefore, can be removed.

Reply:
Okay